mypy/main.py

```diff
@@ -114,10 +114,24 @@ def main(script_path: Optional[str],
         code = 2 if blockers else 1
     if options.error_summary:
         if messages:
+            if options.write_baseline and res:
+                new_errors = 0
+                for errors in res.manager.errors.error_info_map.values():
+                    new_errors += len(
+                        [error for error in res.manager.errors.remove_duplicates(
+                            res.manager.errors.render_messages(
+                                res.manager.errors.sort_messages(
+                                    [error for error in errors if not error.hidden]
+                                )
+                            )
+                        ) if error[3] == "error"]
```

Comment:
what's `error[3]`? can this be turned into a dict or something where the index is more descriptive

Reply:
Nah, it's used extensively internally:
```py
# Type used internally to represent errors:
#   (path, line, column, severity, message, allow_dups, code)
ErrorTuple = Tuple[Optional[str],
                   int,
                   int,
                   str,
                   str,
                   bool,
                   Optional[ErrorCode]]
```